### PR TITLE
drivers: can: add `can_get_mode()` and `can_get_transceiver()` system calls

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
- Add system call `can_get_mode()` for getting the current operation mode of a CAN controller.
- Add system call `can_get_transceiver()` for getting the CAN transceiver associated with a CAN controller.
- Add relevant tests for these two new system calls.
- Use these new system calls in the CAN shell module.

### CODE REVIEW

**Review by alexanderwachter — APPROVED** [review]

**Review by str4t0m — APPROVED** [review]
Thanks, looks good. Just one remark regarding the first commit message.

File: drivers/can/can_stm32_bxcan.c [hunk 1]
```diff
@@ -532,6 +532,8 @@ static int can_stm32_set_mode(const struct device *dev, can_mode_t mode)
 		can->MCR &= ~CAN_MCR_NART;
 	}
 
+	data->common.mode = mode;
+
```
[the remark this answers is not in the document]
Comment (henrikbrixandersen) [reply drivers/can/can_stm32_bxcan.c:536]: Good catch, thanks! I meant to refer to 9051824fa3f1b2d22163e27abef8061c6242464e. Fixed.

File: tests/drivers/can/api/src/transceiver.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * @brief Test getting CAN transceiver device pointer.
+ */
+ZTEST_USER(can_transceiver, test_get_transceiver)
+{
+	const struct device *phy = can_get_transceiver(can_dev);
+
+	zassert_equal(phy, can_phy, "wrong CAN transceiver device pointer returned");
+}
+
+static bool can_transceiver_predicate(const void *state)
```
Comment (martinjaeger) [inline tests/drivers/can/api/src/transceiver.c:29]: Why do we need the `_predicate` if the same checks are also performed in the `_setup` function?

Comment (henrikbrixandersen) [reply tests/drivers/can/api/src/transceiver.c:29]: The predicate failure will cause the entire suite to be skipped. The setup function will not be called then. 

One could argue that we shouldn't test this twice, but it's really for two different purposes.

Comment (martinjaeger) [reply tests/drivers/can/api/src/transceiver.c:29]: Ok, fine with me. Thanks for the explanation.

**Review by martinjaeger — APPROVED** [review]

**Pull request merged**